watchers/configmap_watcher.go

```diff
@@ -0,0 +1,78 @@
+package watcher
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"sync"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+func WatchForChanges(clientset *kubernetes.Clientset, isLocalRun bool, namespace string, mgrConfigMapName string, dataField string, dataValue string, mutex *sync.Mutex) {
+	logger := log.FromContext(context.TODO())
+	logger.V(0).Info("watchForChanges", "namespace", namespace, "data field", dataField, "current value", dataValue)
+	logger.V(1).Info("watchForChanges", "namespace", namespace, "data field", dataField, "current value", dataValue)
+	for {
+		watcher, err := clientset.CoreV1().ConfigMaps(namespace).Watch(context.TODO(),
+			metav1.SingleObject(metav1.ObjectMeta{Name: mgrConfigMapName, Namespace: namespace}))
+		if err != nil {
+			logger.Error(fmt.Errorf("%s", "Unable to create watcher"), "namespace", namespace, "ConfigMap name", mgrConfigMapName)
+			os.Exit(1)
+		}
+		checkConfigMapChanges(watcher.ResultChan(), isLocalRun, dataField, dataValue, mutex)
+	}
+}
+
+func checkConfigMapChanges(eventChannel <-chan watch.Event, isLocalRun bool, dataField string, dataValue string, mutex *sync.Mutex) {
+	for {
+		event, open := <-eventChannel
+		if open {
+			switch event.Type {
+			case watch.Added:
+				fallthrough
+			case watch.Modified:
+				mutex.Lock()
```

Comment:
the pattern for using locks requires `defer mutex.Unlock()` right after the lock is acquired, else, the exit statement on line 59 will break without releasing the lock.